### PR TITLE
Add page Install Baremetal Operator

### PR DESCRIPTION
**Pull request by wrkode**
This PR solves #316 by adding  Install Baremetal Operator page.

### CODE REVIEW

**Comment by tuminoid** [comment]
/cc @lentzi90 @kashifest @Rozzii 
Please take a look.

File: docs/user-guide/src/bmo/install_baremetal_operator.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+1. Clone Metal3 BMO repository `https://github.com/metal3-io/baremetal-operator.git`.
+2. Adapt the configuration settings to your specific needs.
+3. Deploy BMO in the cluster with or without Ironic.
+
+*Note*: This guide assumes that a local clone of the repository is available.
+
+Configuration Settings
+======================
+
+Review and edit the file `ironic.env` found in `config/dev`.
```
Comment (lentzi90) [inline docs/user-guide/src/bmo/install_baremetal_operator.md:14]: Did you mean `config/default`?

File: docs/user-guide/src/bmo/install_baremetal_operator.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+1. Deploying baremetal-operator with Ironic.
+
+2. Deploying baremetal-operator without Ironic.
+
+3. Deploying only Ironic.
+
+## Current structure of baremetal-operator config directory
+
+```diff
```
Comment (lentzi90) [inline docs/user-guide/src/bmo/install_baremetal_operator.md:138]:
```suggestion
```console
```
`diff` is for output of `diff` as far as I know. I would suggest `console` or just `text` for this.

File: docs/user-guide/src/bmo/install_baremetal_operator.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+
+The `config` directory has one top level folder for deployment, namely `default`
+and it deploys only baremetal-operator through kustomization file calling
+`manager` folder. In addition, `basic-auth`, `certmanager`, `crd`, `namespace`,
+`prometheus`, `rbac`, `tls` and `webhook`folders have their own kustomization
+and yaml files. `samples` folder includes yaml representation of sample CRDs.
+
+## Current structure of ironic-deployment directory
+
+```diff
```
Comment (lentzi90) [inline docs/user-guide/src/bmo/install_baremetal_operator.md:223]:
```suggestion
```console
```

**Review by lentzi90 — COMMENTED** [review]
Great work!
Since this is mainly moving/migrating existing content I'm tempted to just take it in as is, but I added a couple of suggestions below.
The markdown linter seems to complain just about the headers and lists so that should hopefully be a quick fix. I.e. it does not allow multiple first level headers, requires consistent styling for them and the lists.

File: docs/user-guide/src/bmo/install_baremetal_operator.md (hunk 1 above)
Comment (wrkode) [reply docs/user-guide/src/bmo/install_baremetal_operator.md:14]: Yes, I did. editing now

Comment (wrkode) [reply docs/user-guide/src/bmo/install_baremetal_operator.md]: @lentzi90, I've commited the canges, could you please review at your best convenience?

Comment (wrkode) [reply docs/user-guide/src/bmo/install_baremetal_operator.md]: @lentzi90 I see that markdown lint is still complaining.
I'm working on it

**Comment by wrkode** [comment]
@lentzi90, I believe markdown lint is ok now. could someone review/test?
PS: again thanks for the patience

**Review by lentzi90 — COMMENTED** [review]
Thank you for working on this!
/lgtm

**Comment by wrkode** [comment]
@Rozzii and @kashifest could you please review and let me know what you think?

**Review by Rozzii — APPROVED** [review]

**Comment by Rozzii** [comment]
Looks good to me but I have no approve right, @kashifest  could you please apporve?

**Review by kashifest — APPROVED** [review]

**Pull request merged**